src/main/java/org/tndata/android/compass/adapter/CategoryFragmentAdapter.java

```diff
@@ -89,62 +115,40 @@ public void onBindViewHolder(final RecyclerView.ViewHolder viewHolder,
-                viewHolder).circleView.getBackground();
-        String colorString = mCategory.getColor();
-        if (colorString != null && !colorString.isEmpty()) {
-            gradientDrawable.setColor(Color.parseColor(colorString));
-        }
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
-            ((CategoryGoalViewHolder) viewHolder).circleView.setBackground
-                    (gradientDrawable);
-        } else {
-            ((CategoryGoalViewHolder) viewHolder).circleView
-                    .setBackgroundDrawable(gradientDrawable);
-        }
-
-        // Set the progress widget for the Goal.
+        ((CategoryGoalViewHolder) viewHolder).descriptionTextView.setText(
+                goal.getDescription());
+        ((CategoryGoalViewHolder) viewHolder).setCircleViewBackgroundColor(
+                mCategory.getColor());
         ((CategoryGoalViewHolder) viewHolder).iconImageView.setImageResource(
                 goal.getProgressIcon());
-
         ((CategoryGoalViewHolder) viewHolder).goalContainer.setOnClickListener(new View
                 .OnClickListener() {
 
             @Override
             public void onClick(View v) {
-                mCallback.chooseBehaviors(goal);
+                ((CategoryGoalViewHolder) viewHolder).toggleCard();
             }
         });
-        ArrayList<Behavior> behaviors = goal.getBehaviors();
-        ((CategoryGoalViewHolder) viewHolder).behaviorContainer.removeAllViews();
-        if (behaviors != null && !behaviors.isEmpty()) {
-            for (final Behavior behavior : behaviors) {
-                BehaviorListView behaviorListView = new BehaviorListView(mContext);
-                behaviorListView.setBehavior(behavior, mCategory);
-                behaviorListView.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View v) {
-                        mCallback.viewBehavior(goal, behavior);
-                    }
-                });
-                ((CategoryGoalViewHolder) viewHolder).behaviorContainer.addView(behaviorListView);
-                if (mContext instanceof Activity) {
-                    for (final Action action : ((CompassApplication) ((Activity) mContext)
-                            .getApplication()).getActions()) {
-                        if ((action.getBehavior() != null && action.getBehavior().equals(behavior))
-                                || (action.getBehavior_id() == behavior.getId())) {
-                            ActionListView actionListView = new ActionListView(mContext);
-                            actionListView.setAction(action);
-
-                            ((CategoryGoalViewHolder) viewHolder).behaviorContainer.addView
-                                    (actionListView);
-
-                        }
-                    }
+
+        // Since the descriptionTextView is outside of the goal_goal_container layout, it
```

Comment:
`viewHolder.itemView` to get the entire card, and make that clickable, rather than description and container